player/lua/select.lua

```diff
@@ -345,6 +345,59 @@ mp.add_key_binding(nil, "select-audio-device", function ()
     })
 end)
 
+mp.add_key_binding(nil, "select-history", function ()
+    local watch_later_dir = mp.get_property("current-watch-later-dir")
+
+    if not watch_later_dir then
+        show_error("No watch later files found.")
+        return
+    end
+
+    local watch_later_files = {}
+
+    for i, file in ipairs(utils.readdir(watch_later_dir, "files") or {}) do
+        watch_later_files[i] = watch_later_dir .. "/" .. file
+    end
+
+    if #watch_later_files == 0 then
+        show_error("No watch later files found.")
+        return
+    end
+
+    local recent_files = {}
+    for _, watch_later_file in pairs(watch_later_files) do
+        local file_handle = io.open(watch_later_file)
+        if file_handle then
+            local line = file_handle:read()
+            if line and line ~= "# redirect entry" and line:find("^#") then
+                recent_files[#recent_files + 1] = {watch_later_file, line:sub(3)}
+            end
+            file_handle:close()
+        end
+    end
+
+    if #recent_files == 0 then
+        show_error("Enable --write-filename-in-watch-later-config to select recent files.")
+        return
+    end
+
+    table.sort(recent_files, function (i, j)
+        return utils.file_info(i[1]).mtime > utils.file_info(j[1]).mtime
+    end)
+
+    for i, recent_file in ipairs(recent_files) do
+        recent_files[i] = recent_file[2]
+    end
+
+    input.select({
+        prompt = "Select a recent file:",
```

Comment:
If I open directory, this will show only one file played as recent from this directory. Also opening this file will open file directly instead of directory.

Reply:
Doesn't --autocreate-playlist load the other files in the directory if that's what you want?

Reply:
I am using this binding, it works well if you don't want to enable `--autocreate-playlist` globally to not apply it to the initially opened file:

`g-w script-binding select/select-watch-later; no-osd set file-local-options/pause yes; no-osd set autocreate-playlist filter; write-watch-later-config`

Dunno if it should just always play the parent directory if not a URL.